### ISSUE TITLE
Question about MultipleNegativesRankingLoss

### ISSUE THREAD

**Issue by Mogady** [UKPLab/sentence-transformers#1142]
Maybe I have a misunderstanding but how does using cross_entropy_loss in MultipleNegativesRankingLoss helps ?

The cross-entropy will only care for the positive example since its' label is 1 while all the other negatives are 0, so the final known equation will be sum(1log(P1)+0log(P2)+0log(P3)) right? so it doesn't matter whether I provide P2/P3 as a hard negative or from the batch this should not affect the loss !!

**Comment by nreimers** [comment]
Scores (probabilities) are softmax normalized. I. e. if P2/P3 are large, because they are hard, P1 will be smaller and the loss larger.

**Comment by liuyukid** [comment]
I don't think the way the sample was constructed for MultipleNegativesRankingLoss was appropriate.

For example, suppose that there are the following relationships：
(a, entailment, b)
(a, entailment, d)
(d, entailment, e)
(a, contradiction, c)
(d, contradiction, f)

Based on this code, 
`train_samples = []
for sent1, others in train_data.items():
    if len(others['entailment']) > 0 and len(others['contradiction']) > 0:
        train_samples.append(InputExample(texts=[sent1, random.choice(list(others['entailment'])), random.choice(list(others['contradiction']))]))
        train_samples.append(InputExample(texts=[random.choice(list(others['entailment'])), sent1, random.choice(list(others['contradiction']))]))`
I can construct the following batch：
(a, b, c)
(e, d, f)

finally,
for a,            (b, d, c, f)     label is 0
But b and d are both  entailment with a

**Comment by nreimers** [comment]
@liuyukid 
Not sure what you mean.

Note, that SNLI and MultiNLI have a special format. For each original sentence A, an entailment, a neutral, and a contradiction example was formed.
